its/src/test/java/com/sonar/it/csharp/UCFGDeserializationTest.java

```diff
@@ -131,6 +135,28 @@ public void read_ucfgs() {
     assertThat(ucfgs).hasSize(926);
   }
 
+  @Test
+  public void ucfgs_not_created_when_rules_not_enabled() throws IOException {
+    // No security rules in QP
+    createQP("S100");
+
+    File projectLocation = new File("projects/SimplCommerce");
+
+    orchestrator.executeBuild(getScannerForMSBuild(projectLocation)
+      .addArgument("begin")
+      .setDebugLogs(true)
+      .setProjectKey(PROJECT_KEY)
+      .setProjectName(PROJECT_KEY)
+      .setProjectVersion("1.0"));
+
+    executeDotNetCore(projectLocation, "build");
+
+    orchestrator.executeBuild(getScannerForMSBuild(projectLocation).addArgument("end"));
```

Comment:
The begin-build-analyse steps are almost identical to the ones in the _rules_enabled_ test. It's probably worth moving them into a help function.